CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+If on Linux, you can run `./gradlew test` from the terminal.
+If on Windows, you can run `gradlew.bat test` from the command prompt.
+If using an IDE like IntelliJ, you can open the Gradle panel, navigate to `MockBukkit-1.xx` -> `Tasks` -> `verification`, then double click `test`.
+
+
+## Committing and Pushing your changes
+
+Once you've completed your changes and confirmed that they work, you'll need to commit, then push your changes.
+To commit your changes, run `git add .` to add all your changes, then `git commit -m "<commit message>"` with `<commit message>` stating what you've done in 70 characters or less.
+Now you're ready to push your changes to GitHub. To do this, run `git push -u origin <branch name>` with `<branch name>` being the name of your branch you created earlier.
+
+
+## Creating a pull request
+
+To create the pull request, head back to your fork on GitHub, and go to the `Pull Requests` tab, and click the `New Pull Request` button.
+On the left, make sure `Base repository` is set to `MockBukkit/MockBukkit` and the branch next to it is set to the latest version (e.g. `v1.19`).
+On the right, make sure `Head repoitory` is set to `Usename/MockBukkit` with `Username` being your GitHub username and the branch being the new branch you created.
+Now you can click the `Create pull request` button to start creating the pull request.
+
+The title of your pull request should be a should description of what you changed, in 70 characters or less.
+In the description box, *please* follow the template provided and fill out everything that's asked for.
+
+Make sure `Allow edits and access to secrets by maintainers` is checked, so we can update and make small tweaks to your pull request, so you don't have to.
+
+If you scroll down, you will see the diff between MockBukkit's source code, and the code you modified.
+If everything in their looks alright, you can click the `Create pull request` button to create the pull request!
+
+Initially GitHub will show that the pull request did not pass checks. If you scroll down to the bottom, you will see a list of all checks and their status.
+If all checks have a green checkmark except for `Check PR has release/* label`, you're good to go, and it will succeed as soon as a maintainer assigns a label to your pull request.
+
+If the `Build` check is failing, this means that there was an error when compiling your code or a test failed.
+To see why this happened, you can click `Details` then `build` to see the Gradle log. If you're unsure of how to fix it, you can wait for a maintainer to help.
+
+A few minutes after you've created your pull request, `SonarCloud` will leave a comment on the quality of your code.
+If everything is `0`, you're good to go! If SonarCloud has issues, you can click on them, and it will show you what's wrong and how you should fix it.
+Usually these are reasonable, and we would appreciate if you resolved them.
```

Comment:
We should also add a section about the "Issues" tab.
Since a good bug report is just as much of a contribution as a good PR.